spec/index.bs

```diff
@@ -2148,21 +2148,88 @@ premultiplication) in which the WebGPU numeric values are to be interpreted.
 WebGPU allows all of the color spaces in the {{PredefinedColorSpace}} enum.
 Note, each color space is defined over an extended range, as defined by the referenced CSS definitions,
 to represent color values outside of its space (in both chrominance and luminance).
+Any numeric value in any color space is a well-defined "color" according to its extended color
+space definition, regardless of whether the point represents a real or imaginary color
+(WebGPU makes no distinction between real and imaginary colors).
+That "color" can be represented in other color spaces as well, where it may be in-gamut.
 
-An <dfn dfn>out-of-gamut premultiplied RGBA value</dfn> is one where any of the R/G/B channel values
-exceeds the alpha channel value. For example, the premultiplied sRGB RGBA value [1.0, 0, 0, 0.5]
-represents the (unpremultiplied) color [2, 0, 0] with 50% alpha, written `rgb(srgb 2 0 0 / 50%)` in CSS.
-Just like any color value outside the sRGB color gamut, this is a well defined point in the extended color space
-(except when alpha is 0, in which case there is no color).
-However, when such values are output to a visible canvas, the result is undefined
-(see {{GPUCanvasAlphaMode}} {{GPUCanvasAlphaMode/"premultiplied"}}).
+<h4 data-dfn-type=dfn>Out-of-Range Premultiplied RGBA Values
+<span id=out-of-gamut-premultiplied-rgba-value><!-- historical permalink --></span>
+</h4>
+
+An [=out-of-range premultiplied RGBA value=] is one for which the unpremultiplied representation
+of the same color would be outside of the range of the format.
+
+<div class=example>
+    For example, on a display with a color space of exactly Display P3,
+    the unpremultiplied sRGB RGBA value `[1.04, 0, 0, 0.5]` and the
+    premultiplied sRGB RGBA value `[0.52, 0, 0, 0.5]` both represent the
+    CSS color <code><a funcdef>color</a>(srgb 1.04 0 0 / 0.5)</code>, which is equivalent to the
+    in-gamut Display P3 color <code><a funcdef>color</a>(display-p3 0.95 0.21 0.15 / 0.5)</code>.
+
+    When represented in a {{PredefinedColorSpace/"srgb"}} {{GPUTextureFormat/"rgba8unorm"}} canvas,
+    both values are out-of-range for the purposes of this definition,
+    even though the premultiplied value is representable.
+
+    (The colors in this example are in the gamut of the display to avoid any gamut mapping.)
+</div>
+
+When a canvas containing such a value is
+[$get a copy of the image contents of a context|used as an image source$],
+the returned image must represent the original color values without clamping,
+regardless of the canvas's {{GPUCanvasConfiguration/format}}.
+
+However, when a ({{GPUCanvasAlphaMode/"premultiplied"}}) canvas containing such a value is
+displayed *on screen*, the visible result for the entire canvas is undefined.
```

Comment:
Checking that I understand.

Consider a canvas that is premultiplied, `rgba16float`, `srgb` being displayed on a P3 monitor.

If the value `[0.52, 0, 0, 0.5]` is written to the canvas, and the canvas is displayed, then the result is undefined even though it's within the gamut of the display, because of various machinations that might happen throughout the display pipeline.

This seems reasonable to me. Would the idea be to make this valid when in extended mode?